destination_test.go

```diff
@@ -1,15 +1,378 @@
+		}
+	}
+
+	sdk.Logger(ctx).Info().Msg("successfully cleaned test files")
+	sdk.Logger(ctx).Info().Msg("deleting vector stores...")
+
+	vectorStores, err := client.ListVectorStores(ctx, openai.Pagination{})
+	if err != nil {
+		sdk.Logger(ctx).Fatal().Err(err).Msg("failed to list vector stores")
+	}
+
+	for _, vectorStore := range vectorStores.VectorStores {
+		_, err := client.DeleteVectorStore(ctx, vectorStore.ID)
+		if err != nil {
+			sdk.Logger(ctx).Fatal().Err(err).Str("vectorStoreID", vectorStore.ID).Msg("failed to delete vector store")
+		}
+	}
+
+	sdk.Logger(ctx).Info().Msg("successfully deleted test vector stores")
+}
+
+type testFile struct {
+	name     string
+	contents string
+}
+
+func newTestFile() testFile {
+	return testFile{
+		name:     fmt.Sprint("test-", uuid.NewString()[0:8], ".txt"),
+		contents: "test contents",
+	}
+}
+
+func (f testFile) recCreate() opencdc.Record {
+	return opencdc.Record{
+		Position:  nil,
+		Operation: opencdc.OperationCreate,
+		Metadata:  nil,
+		Key:       opencdc.RawData(f.name),
+		Payload: opencdc.Change{
+			After: opencdc.RawData(f.contents),
+		},
+	}
+}
+
+func (f testFile) recSnapshot() opencdc.Record {
+	return opencdc.Record{
+		Position:  nil,
+		Operation: opencdc.OperationSnapshot,
+		Metadata:  nil,
+		Key:       opencdc.RawData(f.name),
+		Payload: opencdc.Change{
+			After: opencdc.RawData(f.contents),
+		},
+	}
+}
+
+func (f testFile) recUpdate() opencdc.Record {
+	return opencdc.Record{
+		Position:  nil,
+		Operation: opencdc.OperationUpdate,
+		Metadata:  nil,
+		Key:       opencdc.RawData(f.name),
+		Payload: opencdc.Change{
+			After: opencdc.RawData(f.contents),
+		},
+	}
+}
+
+func assertFileWrittenAndUnique(
+	ctx context.Context, t *testing.T, writtenFile testFile, vectorStoreID string,
+) {
+	t.Helper()
+	client := testClient(t)
+	is := is.New(t)
+
+	// assert that file exists
+	files, err := client.ListFiles(ctx)
+	is.NoErr(err)
+
+	var fileID string
+	for _, openaiFile := range files.Files {
+		if openaiFile.FileName == writtenFile.name {
+			if fileID != "" {
+				t.Fatalf("found multiple files with name %s", writtenFile.name)
+			}
+			fileID = openaiFile.ID
+		}
+	}
+	if fileID == "" {
+		t.Fatalf("failed to find filename %s", writtenFile.name)
+	}
+
+	// OpenAI doesn't provide a way to get the file contents, so we can't assert
+	// that the file contents are correct. In theory we could create an AI agent and
+	// check that the file that it has access to has the proper contents.
+
+	_, err = client.RetrieveVectorStoreFile(ctx, vectorStoreID, fileID)
+	is.NoErr(err)
+}
+
+func assertDeletedFile(ctx context.Context, t *testing.T, filename string) {
+	t.Helper()
+	client := testClient(t)
+	is := is.New(t)
+
+	files, err := client.ListFiles(ctx)
+	is.NoErr(err)
+
+	for _, openaiFile := range files.Files {
+		if openaiFile.FileName == filename {
+			t.Fatalf("file %s still exists", filename)
+		}
+	}
+}
+
+func envGetter(key string) func(t *testing.T) string {
```

Comment:
Could we use `setEnv` instead? This is a code snippet preserving a possible original value and setting one for testing purposes.

e.g.:

 ```golang
    // Save the original value of the environment variable
    originalValue := os.Getenv("MY_ENV_VAR")

    // Set the environment variable for the test
    if err := os.Setenv("MY_ENV_VAR", "test_value"); err != nil {
        t.Fatalf("Failed to set environment variable: %v", err)
    }

    // Defer resetting the environment variable to its original value
    defer func() {
        if err := os.Setenv("MY_ENV_VAR", originalValue); err != nil {
            t.Fatalf("Failed to reset environment variable: %v", err)
        }
    }()
```

Reply:
I don't think setting the openai api key env var will be useful here, as I'm not mocking the api client. I don't think it will work with an invalid api key either.

I agree though that `envGetter` might be too much for getting a single env var, we could just this:

```go
func getenv(t *testing.T, key string) string {
	t.Helper()
	env := os.Getenv(key)
	if env == "" {
		t.Fatalf("missing env var %s", key)
	}

	return env
}
```